### PR TITLE
feature(visualize) avoid overplotting in parameter visualization

### PR DESCRIPTION
**Pull request by FFroehlich**
Overplotting in `visualize.parameters` can be quite severe when looking at results from a larger number of starts. In this implementation I rescaled the `alpha` according to `min(1,5/clustsize)`. Not 100% happy with the results, any suggestions for improvement are welcome.
 
Example with ~800 starts and cluster sizes ~50 and ~750

Before:
before

After:
after

### CODE REVIEW

File: pypesto/visualize/clust_color.py [hunk 1]
```diff
@@ -81,6 +81,18 @@ def assign_clustered_colors(vals):
...
     cols_clust[vmax] = (0.7, 0.7, 0.7, 1)
 
+    for icluster in range(vmax+1):
+        if icluster == vmax:
+            # grey color for 1-size clusters as the last color in 'cols_clust'
+            # alpha set according to number of 1-size clusters (+1 to avoid
+            # zero division)
+            cols_clust[icluster][3] = min(1, 5 / (sum(clustsize == 1) + 1))
+        else:
+            # normalize alpha according to clustersize (why is this in inverse
```
[the remark this answers is not in the document]
Comment (DantongWang) [reply pypesto/visualize/clust_color.py]: sorry, do you mean the numbers in `clustsize` array is inverse order or what?

[the remark this answers is not in the document]
Comment (yannikschaelte) [reply pypesto/visualize/clust_color.py]: I guess the ``clustsize[vmax - 1 - icluster]`` looks strange, maybe it's possible to have ``cols_clust[icluster][3] = ... clustsize[icluster]`` instead, i.e. have both in the same order?

[the remark this answers is not in the document]
Comment (DantongWang) [reply pypesto/visualize/clust_color.py]: ah... I get it, I think it has something to do with the automatic clustering process, which I don't completely understand what is happening there. 
Bad news: it is not always in inverse order, but kind of 'random' order... Will need to think a better way to assign alpha.

[the remark this answers is not in the document]
Comment (DantongWang) [reply pypesto/visualize/clust_color.py]: I am trying to change the `assign_cluster` part to fix this problem. will update in a new branch later.

**Comment by FFroehlich** [comment]
> 1. I'd make the alpha reduction optional. In particular when there are just few starts, it may not be necessary. In addition, some printers tend to not like un-full colors. Maybe some option `balance_alpha=False` or sth like that?

 The `5` in `min(1,5/clustsize)` should ensure that the alpha reduction only becomes noticable for clusters with more than ~10 elements. Nevertheless I have made it optional.

> 2. You probably looked into this already, but is there a way to correlate alpha also with the quality of the cluster's function value? I could imagine that when 99% found the global optimum, and 1% is off, the latter might look like more important because they are are less opaque. For a start, one might think about not reducing alpha for the optimal cluster?

Not sure. For this example I was particularly interested in reducing the alpha for the optimal cluster. I guess in that case it would probably best to turn alpha balancing off?

**Comment by yannikschaelte** [comment]
let's maybe wait with this until #114 is merged, i think this was addressed there partly as well

**Comment by paulstapor** [comment]
Okay, things should be merged now. I used @yannikschaelte 's branch merge_plotting, to clean these things up. I would prefer merging this branch now ans using merge_plotting, to add some visualization examples in notebooks for the new functionalities (preferably the rosenbrock example notebook, since this doesn't need amici and I would like to keep dependencies as simple as possible for example notebooks)

**Comment by yannikschaelte** [comment]
closing this one since changes incorrectly displayed. reopened in #135, which is identical.

**Pull request closed without merging**